### PR TITLE
Add API permissions to the model endpoints

### PR DESCRIPTION
**Pull request by slimjim777**

### CODE REVIEW

File: service/utils.go [hunk 1]
```diff
@@ -198,3 +199,23 @@ func validateModelName(name string) error {
 
 	return nil
 }
+
+// checkUserPermissions retrieves the user from the JWT.
+// The user will be restricted by the accounts the username can access and their role.
+// If user authentication is turned off, the JWT will irrelevant.
+func checkUserPermissions(w http.ResponseWriter, r *http.Request) (string, error) {
```
Comment (rmescandon) [inline service/utils.go]: still thinking for a method name...
getAllowedUser(), getAuthUser()...
checkUserPermissions() sounds to me that you provide the permissions that are checked against something. Here is the token what is checked and from there it is extracted the allowed user (or no user if * is allowed)

I could accept checkUser() even as valid

Comment (slimjim777) [inline service/utils.go]: I think I've added a check that we have an admin user in the next PR :)

File: datastore/usermanager.go [hunk 2]
```diff
@@ -209,3 +217,32 @@ func rowsToUsers(rows *sql.Rows) ([]User, error) {
 
 	return users, nil
 }
+
+func (db *DB) checkUserPermissions(username string) int {
```
Comment (slimjim777) [inline datastore/usermanager.go]: This gets changed in the next PR! Can we let this through here and I'll take the naming into account there.

Comment (rmescandon) [inline datastore/usermanager.go]: sure.

**Pull request merged**